flutter/ARCHITECTURE.md

```diff
@@ -521,56 +481,46 @@ such as server URLs.
-build and deploy the apps. All these phases are detailed explained in the 
-[release workflow script](.github/workflows/release.yml).
-
-The Github Actions triggers the script that begins the release process, but the heavy work is made by using 
-[Fastlane](https://fastlane.tools/) automation tool. Fastlane gives us a set of tools to automate all the heavy work
-of configuring a project environment to be deployed, build and deploy it to production. Every the release script from
-Github Actions finish the environment setup, we use Fastlane to run the targets tests, analyzer, bump the build version
-from the targets, build the projects and finally deploy them. We currently have Fastlane running for iOS and Android 
-targets, and the lanes that perform such tasks are described in details in both platforms Fastfile's:
-
-- [iOS Fastfile](ios/fastlane/Fastfile)
-- [Android Fastfile](android/fastlane/Fastfile)
-
-The main difference between the mobile platforms is that Android build number is indefinitely incremented, since it's
-not allowed to reset an Android project build number, while iOS resets its build number every time the app version 
-changes. 
-
-You may notice that both platforms Fastfiles are quite different. This happens firstly because both platforms 
-environments are completely different - as an example of that, iOS requires provision profiles and certificates to 
-build and deploy the application, while Android requires a sign key to before deploying the app. They also differ 
-because there are some actions being used in iOS Fastlane that doesn't exists on Android - an example of that is that
-the iOS Fastlane uses an action to fetch the last used build number on TestFlight so it can increment it and deploy the
-next version, while Android just trust in the number that exists in the local `build.gradle`.
\ No newline at end of file
+The [release workflow](../.github/workflows/release-flutter.yml) is triggered each time a new Github tag is published,
+running the deploy process to all supported platforms. It prepares the environment to build and release all app targets
+by setting up Flutter environment, private information (like distribution certificates from an external repository) and
+other phases that are required to deploy Memo.
+
+Under the hood, all of the heavy work is done by [Fastlane](https://fastlane.tools/). In simple terms, this tool
+automates the build and deploy to the platform's environment. After the deploy is completed, it bumps the build version
+from the targets. We currently have Fastlane running for [iOS](ios/fastlane/Fastfile) and
+[Android](android/fastlane/Fastfile) targets.
+
+> There's a catch to both iOS and Android build-versioning: the Android's build number is indefinitely incremented,
+> since it's not possible to reset an Android project build number, while iOS resets its build number every time a new 
+> version (major, minor or patch, doesn't matter) is bumped.
```

Comment:
I don't think this comment aggregates in the document, since this is basically knowledge about Android versioning

Reply:
It gives context to those who don't know about such problem, and that's why this doc exist, to explain why - as it's not a technical code documentation, not even mentioning that assuming most of the developers know about this.

This is a document to explain the decisions, so that we don't forget and come back to discuss things like, in this case, "why don't we use the same build number for both platforms and improve consistency between them?".

Reply:
I was trying to say is that this is not a problem, and we not even changed that. It is working as it is supposed to be.

As a comparison example, I think it's worth it to mention when we remove `version` from `pubspec.yaml` when both iOS and Android targets have different versions, or like when we modify the Flutter generated `Podfile` to add some optimization or stuff like that, because in these situations we changed the default way of how thinks worked due to some project specific requirements.

This versioning docs just states what happens by default: iOS resets in every new version while Android versions the build incrementally. This happens even in native projects. We haven't changed that, that's why I think that this docs are not aggregating anything, they are just for educational purposes.

Reply:
Yeah, I got your point, but like I've said, stating the obvious may not be obvious for everyone. This is a contextual knowledge that may teach something to people that didn't knew about that.

Think about some iOS developer that never adventured in the Android world - and vice-versa. They could be experienced developers, although they lack this definition, and adding here was pertinent enough to make them understand why these platform's version differ from each other and how we version the application, given this condition.

Reply:
I get it, but where is the limit between contextual knowledge and we actually documented extra-stuff that is specific context knowledge? In my point of view, we should assume some knowledge from the public when writing the `ARCHITECTURE.md`, and highlight the decisions that may differ from other Flutter projects and specific changes that we've made to the ecosystem, like updating the Podfile or creating different iOS schemas, as examples. If not, we may assume that the public knows barely nothing about the Flutter ecosystem and explain everything. I know that this is not your point of view, I'm just making an absurd example to question: where the boundary lives? Again, in my opinion, the boundary are decisions that may be different in other Flutter projects and ecosystem changes that we made, and versioning is not one of them IMO.

I'm ok with the decision of keeping it, I just personally think it doesn't aggregates anything, but I may be wrong here. I leave this decision to you since I'm ok with both scenarios, just wanted to state what I think that should be in this document and what I think it shouldn't.

Reply:
> we should assume some knowledge from the public when writing the `ARCHITECTURE.md`

Definitely, but this is far too much to expect from anyone. This assumes that someone has knowledge on the deploy process. There may be people who are not directly in charge of this process in their respective projects, or even they haven't had any production projects - they simply never bothered to go through it. People may also come from Flutter/mobile from other non-traditional backgrounds, such as desktop, IoT, etcetera.

> where the boundary lives?

In all honesty, there is no such thing. This is too subjective and, even if we delimit the edges of this, like "when a developer that has worked with X has probably faced that", it'll be blurry anyways.

> it doesn't aggregates anything,

It does, but not for you, as you already know about this.

Reply:
The exact same logic occurs with your https://github.com/olmps/memo/pull/195#discussion_r716861246 suggestion.

One should know that this is something to expect if you have knowledge on the Apple development ecosystem - but it's not true for everyone, because just like I've above, people come from different backgrounds and have different experiences.